harness/determined/estimator/_estimator_trial.py

```diff
@@ -23,6 +24,47 @@
 VERY_LARGE_NUMBER = 9999999999999999
 
 
+class RNGStateHook(estimator.RunHook):
+    def __init__(self, checkpoint_dir: pathlib.Path):
+        try:
+            self.rng_state = None
+            with open(checkpoint_dir.joinpath("rng_state.pkl"), "rb") as f:
+                self.rng_state = pickle.load(f)
+        except IOError:
+            logging.debug("No RNG state found in checkpoint_dir")
+
+    def after_create_session(self, session, coord):  # type: ignore
+        if self.rng_state is not None:
```

Comment:
There won't always be an RNG state, because it might be an old checkpoint, and of course after_create_session is also called on new experiments. So I was using the existence of rng_state as the signal for that. If it doesn't exist, I already log a warning. Is there a better signal to use?

Reply:
I was thinking `self.rng_state` should be initialized when creating an estimator trial controller but I saw in the latest version you have removed this internal state so it looks good to me.